### PR TITLE
Transports can now unload only when on a valid unload tile.

### PR DESCRIPTION
**Pull request by ThatIsAUsername**

### CODE REVIEW

File: src/Engine/Utils.java [hunk 1]
```diff
@@ -69,15 +69,16 @@
   {
     ArrayList<XYCoord> locations = findLocationsInRange(map, moveLoc, 1);
     ArrayList<XYCoord> dropoffLocations = new ArrayList<XYCoord>();
-    for( XYCoord loc : locations )
-    {
-      // Add any location that is empty and supports movement of the cargo unit.
-      if( (map.isLocationEmpty(loc) || map.getLocation(loc).getResident() == transport)
-          && cargo.model.movePower >= cargo.model.propulsion.getMoveCost(map.getEnvironment(loc.xCoord, loc.yCoord)) )
+    if( cargo.model.propulsion.canTraverse(map.getEnvironment(moveLoc)) )
```
Comment (ThislsAUsername) [inline src/Engine/Utils.java]: That's awfully idiomatic. I like it.

**Comment by ThislsAUsername** [comment]
Looks good, sans testing.

**Comment by ThatIsAUsername** [comment]
> Looks good, sans testing.
> (sans testing because my repo is neck-deep in breaking everything so I can fix it again)

Just make another copy of the repo. I've been known to have... too many copies, sometimes spread across multiple computers. Good for quick tests/bugfixes while maintaining an involved dev branch.
Or don't, I ain't your boss.

**Comment by ThislsAUsername** [comment]
> 
> 
> > Looks good, sans testing.
> > (sans testing because my repo is neck-deep in breaking everything so I can fix it again)
> 
> Just make another copy of the repo. I've been known to have... too many copies, sometimes spread across multiple computers. Good for quick tests/bugfixes while maintaining an involved dev branch.
> Or don't, I ain't your boss.

I dunno if I can afford that, man. I only have 66 gigs left 😛

**Pull request merged**